bytecomp/matching.ml

```diff
@@ -13,7 +13,56 @@
 (*                                                                        *)
 (**************************************************************************)
 
-(* Compilation of pattern matching *)
+(* Compilation of pattern matching
+
+   Based upon Lefessant-Maranget ``Optimizing Pattern-Matching'' ICFP'2001.
+
+   A previous version was based on Peyton-Jones, ``The Implementation of
+   functional programming languages'', chapter 5.
+
+   # Overview of the implementation
+
+     (split_and_precompile)
+   We first split the matching along its first column -- simplifying pattern
+   heads in the process --, so that we obtain a sorted list of pattern matchings
```

Comment:
The end of this paragraph seems to suggest that the name "pattern matching" is defined to capture the structure you get after splitting&precompiling. This is not quite right, as the non-split matchings are also represented by the `pattern_matching` type.

Reply:
> The end of this paragraph seems to suggest that the name "pattern matching" is defined to capture the structure you get after splitting&precompiling.

That was not my intention.